plugins/woocommerce/src/Blocks/BlockTypes/AbstractInteractivityAPIBlock.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\Blocks\BlockTypes;
+
+/**
+ * AbstractInteractivityAPIBlock class.
+ */
+abstract class AbstractInteractivityAPIBlock extends AbstractBlock {
```

Comment:
Considering that the iAPI is a feature that we can add/remove to/from a block and not a block type specialization I think a better fit for this concept is to use a trait instead of an abstract class. This will allow us to plugin or out this specific block support and others if we follow the composition pattern instead of the inheritance one for similar cases.

Reply:
A trait would be fine maybe as a final design, but for this it's really done to handle technical debt where we support the inherited methods of AbstractBlock

Reply:
The editor will still use scripts, it's just the frontend part that will use script modules. Therefore, maybe a better name (more generic) would be just `AbstractInteractiveBlock`.

Reply:
@luisherranz Yep totally I can get behind that.